routers/web/web.go

```diff
@@ -1042,13 +1049,20 @@ func RegisterRoutes(m *web.Route) {
 					m.Post("/{lid}/unlock", repo.LFSUnlock)
 				})
 			})
-
-			m.Group("/runners", func() {
-				m.Get("", repo.Runners)
-				m.Combo("/{runnerid}").Get(repo.RunnersEdit).
-					Post(web.Bind(forms.EditRunnerForm{}), repo.RunnersEditPost)
-				m.Post("/{runnerid}/delete", repo.RunnerDeletePost)
-				m.Get("/reset_registration_token", repo.ResetRunnerRegistrationToken)
+			m.Group("/actions", func() {
+				m.Get("", repo_setting.RedirectToDefaultSetting)
+				m.Group("/runners", func() {
+					m.Get("", repo_setting.Runners)
+					m.Combo("/{runnerid}").Get(repo_setting.RunnersEdit).
+						Post(web.Bind(forms.EditRunnerForm{}), repo_setting.RunnersEditPost)
+					m.Post("/{runnerid}/delete", repo_setting.RunnerDeletePost)
+					m.Get("/reset_registration_token", repo_setting.ResetRunnerRegistrationToken)
+				})
+				m.Group("/secrets", func() {
+					m.Get("", repo_setting.Secrets)
+					m.Post("", web.Bind(forms.AddSecretForm{}), repo_setting.SecretsPost)
+					m.Post("/delete", repo_setting.DeleteSecret)
+				})
```

Comment:
It's better to use something like `addWebhookEditRoutes` to avoid these duplicate code blocks

Reply:
The secrets related routes are not exactly the same, for example, secret post pass different params to `PerformSecretsPost`  function for different settings

<img width="300" alt="Screen Shot 2023-04-27 at 10 21 39" src="https://user-images.githubusercontent.com/17645053/234743655-45f26c40-5fc9-4d49-b628-52c9f6787b4a.png"><img width="300" alt="Screen Shot 2023-04-27 at 10 22 21" src="https://user-images.githubusercontent.com/17645053/234743663-9b1dad2f-52fd-49ab-8d60-da47644fc696.png"><img width="300" alt="Screen Shot 2023-04-27 at 10 22 37" src="https://user-images.githubusercontent.com/17645053/234743666-abaf7afb-e563-40d5-b588-f564af094afa.png">

Reply:
I see `addWebhookEditRoutes` passed data to different levels, i will modify this with reference to `addWebhookEditRoutes`

Reply:
Done in https://github.com/go-gitea/gitea/pull/24200/commits/fa112e5bb64885fe9943d57ab08d97895e9a3934, please review again

Reply:
> It's better to use something like `addWebhookEditRoutes` to avoid these duplicate code blocks

Done in https://github.com/go-gitea/gitea/pull/24200/commits/fa112e5bb64885fe9943d57ab08d97895e9a3934, please review again, thanks